packages/amplify_datastore/example/integration_test/separate_integration_tests/has_one_explicit_child_test.dart

```diff
@@ -0,0 +1,200 @@
+/*
+ * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import 'package:amplify_datastore/amplify_datastore.dart';
+
+import 'package:integration_test/integration_test.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:amplify_flutter/amplify_flutter.dart';
+
+import '../utils/setup_utils.dart';
+import '../utils/wait_for_expected_event_from_hub.dart';
+import 'models/has_one/ModelProvider.dart';
+
+void main() {
+  IntegrationTestWidgetsFlutterBinding.ensureInitialized();
+
+  group('HasOne (parent refers to child with explicit connection field)', () {
+    // schema
+    // type HasOneParent @model {
+    //   id: ID!
+    //   name: String
+    //   explicitChildID: ID
+    //   explicitChild: HasOneChild @hasOne(fields: ["explicitChildID"])
+    // }
+    // type HasOneChild @model {
+    //   id: ID!
+    //   name: String
+    // }
+    final enableCloudSync = shouldEnableCloudSync();
+    var child = HasOneChild(name: 'child');
+    // Curretly with @hasOne, parent -> child relationship is created
+    // by assign child.id to the connection field of the parent
+    var parent =
+        HasOneParent(name: 'HasOne (explicit)', explicitChildID: child.id);
+    late Future<SubscriptionEvent<HasOneChild>> childEvent;
+    late Future<SubscriptionEvent<HasOneParent>> parentEvent;
+
+    setUpAll(() async {
+      await configureDataStore(
+          enableCloudSync: enableCloudSync,
+          modelProvider: ModelProvider.instance);
+
+      childEvent = Amplify.DataStore.observe(HasOneChild.classType).first;
+      parentEvent = Amplify.DataStore.observe(HasOneParent.classType).first;
+    });
+
+    testWidgets('precondition', (WidgetTester tester) async {
+      var queriedChildren =
```

Comment:
Would it be possible to have:
a) utilities for testing commonly repeated things like "precondition" "save child" "save parent" etc... as much of the code seems copy/pasted with small variations
b) a high-level utility, like `testCommonRelationshipOperations({bool saveChild = true, bool saveParent = true, ....})` that each of these tests could call, then just provide false for the operations they don't need to test

Higher-level goal to save some electricity/repetition here and make tests easier to maintain.

Reply:
Added reusable test util function to simplify the scripts in each test suite.